ee/tpmrunner/tpmrunner.go

```diff
@@ -0,0 +1,249 @@
+// creating and peristing a new one if needed
+func (tr *tpmRunner) Execute() error {
+	durationCounter := backoff.NewMultiplicativeDurationCounter(time.Second, time.Minute)
+	retryTicker := time.NewTicker(durationCounter.Next())
+	defer retryTicker.Stop()
+
+	for {
+		// try to create signer if we don't have one
+		if tr.signer == nil {
+			ctx := context.Background()
+			if err := tr.loadOrCreateKeys(ctx); err != nil {
+				tr.slogger.Log(ctx, slog.LevelError,
+					"loading or creating keys in execute loop",
+					"err", err,
+				)
+			}
+		}
+
+		if tr.signer != nil {
+			retryTicker.Stop()
+		}
+
+		select {
+		case <-retryTicker.C:
+			retryTicker.Reset(durationCounter.Next())
+			continue
+		case <-tr.interrupt:
+			tr.slogger.Log(context.TODO(), slog.LevelDebug,
+				"interrupt received, exiting secure enclave signer execute loop",
+			)
+			return nil
+		}
+	}
+}
+
+func (tr *tpmRunner) Interrupt(_ error) {
+	// Only perform shutdown tasks on first call to interrupt -- no need to repeat on potential extra calls.
+	if tr.interrupted.Load() {
+		return
+	}
+
+	tr.interrupted.Store(true)
+
+	// Tell the execute loop to stop checking, and exit
+	tr.interrupt <- struct{}{}
+}
+
+// Public returns the public hardware key
+func (tr *tpmRunner) Public() crypto.PublicKey {
+	if tr.signer != nil {
+		return tr.signer.Public()
+	}
+
+	if err := tr.loadOrCreateKeys(context.Background()); err != nil {
+		tr.slogger.Log(context.Background(), slog.LevelError,
+			"loading or creating keys in public call",
+			"err", err,
+		)
+
+		return nil
+	}
+
+	return tr.signer.Public()
+}
+
+func (tr *tpmRunner) Type() string {
+	return "tpm"
+}
+
+func (tr *tpmRunner) Sign(rand io.Reader, digest []byte, opts crypto.SignerOpts) ([]byte, error) {
+	if tr.signer == nil {
+		return nil, errors.New("no signer available")
+	}
+
+	return tr.signer.Sign(rand, digest, opts)
+}
+
+// This duplicates some of pkg/osquery/extension.go but that feels like the wrong place.
+// Really, we should have a simpler interface over a storage layer.
+const (
+	privateEccData = "privateEccData"
+	publicEccData  = "publicEccData"
+)
+
+func fetchKeyData(store types.Getter) ([]byte, []byte, error) {
+	pri, err := store.Get([]byte(privateEccData))
+	if err != nil {
+		return nil, nil, err
+	}
+
+	pub, err := store.Get([]byte(publicEccData))
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return pri, pub, nil
+}
+
+func storeKeyData(store types.Setter, pri, pub []byte) error {
+	if pri != nil {
+		if err := store.Set([]byte(privateEccData), pri); err != nil {
+			return err
+		}
+	}
+
+	if pub != nil {
+		if err := store.Set([]byte(publicEccData), pub); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// clearKeyData is used to clear the keys as part of error handling around new keys. It is not intended to be called
+// regularly, and since the path that calls it is around DB errors, it has no error handling.
+// nolint:unused
```

Comment:
I think you can remove the nolint directive here

Reply:
done